tests/productions/test_p8.py

```diff
@@ -0,0 +1,311 @@
+import unittest
+from random import choice
+
+from matplotlib import pyplot
+from networkx import Graph
+
+from agh_graphs.productions.p1 import P1
+from agh_graphs.productions.p2 import P2
+from agh_graphs.productions.p8 import P8
+from agh_graphs.utils import gen_name, add_interior, get_neighbors_at, find_overlapping_vertices, join_overlapping_vertices, \
+    get_node_at, centroid
+from agh_graphs.visualize import visualize_graph_3d, visualize_graph_layer
+
+
+class P8Test(unittest.TestCase):
+
+    def testHappyPath(self):
+        graph = createCorrectGraph()
+        prod_input = [x for x, y in graph.nodes(data=True) if y['label'] == 'I']
+        P8().apply(graph, prod_input)
+        visualize_graph_3d(graph)
```

Comment:
Ostatnia rzecz jaką zauważyłem że brakuje to if'y wokół wizualizacji testów: by default nie powinniśmy pokazywać wykresów. Od tego jest zmienna `tests.utils.visualize_tests`. To jest to miejsce + plus każde inne w testach.